codegen-renderers/src/main/kotlin/io/vrap/codegen/languages/typescript/client/files_producers/ClientFileProducer.kt

```diff
@@ -57,11 +57,12 @@ export type VFile = {
 """)
 
 fun produceClient() = TemplateFile(relativePath = "base/requests-utils.ts", content = """
+
 import { Middleware, MiddlewareArg, CommonRequest } from "./common-types";
 
-export class ApiRequest<I, O, T extends CommonRequest<I>> {
+export class ApiRequest<O> {
```

Comment:
I didn't know that we used so much type parameters here... And I prefer to have as few as possible type parameters. That means i like this change a lot 👍

Reply:
among all these parameters on the one left help in type completion because it signals the return type of the promise this is why it was the only one left.